server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -194,7 +195,8 @@ private void initialization() throws Exception {
     }
     grpcPort = shuffleServerConf.getInteger(ShuffleServerConf.RPC_SERVER_PORT);
     nettyPort = shuffleServerConf.getInteger(ShuffleServerConf.NETTY_SERVER_PORT);
-    if (nettyPort > 0) {
+    if (nettyPort >= 0) {
+      // when nettyPort is zero,actual netty port will be changed,but id can't be change.
```

Comment:
It's not a good idea to use zero here.

Reply:
yes,it's not a good idea,but it's work. I think id gen with 'ip+grpcPort+proccessid' format whern use zero is better or do you have any other proposal?

Reply:
It don't work. Client will use the Netty port of shuffle server id.

Reply:
It seems that the `nettyPort` won't be zero because of Line 131.